.repo/sample-list-generator/src/prepare-chrome-types.ts

```diff
@@ -0,0 +1,68 @@
+import fetch from 'node-fetch';
+import path from 'path';
+import fs from 'fs/promises';
+import { ExtensionApiMap } from './types';
+
+const SINGULAR_TO_PLURAL_MAP: Record<string, string> = {
+  property: 'properties',
+  method: 'methods',
+  event: 'events',
+  type: 'types'
+};
+
+// get variable type
+// e.g. method-openPopup -> methods
+const getTypeFromPageId = (pageId: string) => {
+  return SINGULAR_TO_PLURAL_MAP[pageId.split('-').shift() as string];
+};
+
+// Fetch the latest version of the chrome types from storage
+const fetchChromeTypes = async (): Promise<Record<string, any>> => {
+  console.log('Fetching chrome types...');
+
+  const response = await fetch(
+    'https://storage.googleapis.com/download/storage/v1/b/external-dcc-data/o/chrome-types.json?alt=media'
+  );
+  const chromeTypes = await response.json();
+  return chromeTypes;
+};
+
+const run = async () => {
+  const result: ExtensionApiMap = {};
+
+  const chromeTypes = await fetchChromeTypes();
+
+  for (const [chromeApiKey, chromeApiDetails] of Object.entries(chromeTypes)) {
+    const apiDetails: ExtensionApiMap[string] = {
+      properties: [],
+      methods: [],
+      types: [],
+      events: []
+    };
+
+    for (let variable of chromeApiDetails._type.properties) {
+      const name = variable.name as string;
+      const type = getTypeFromPageId(variable._pageId) as string;
```

Comment:
Can we just use `variable.kind` here? I think the mapping is here: https://unpkg.com/typedoc@0.24.8/dist/lib/models/reflections/kind.d.ts

Reply:
I think `variable.kind` may not be sufficient to determine the type of API in the doc. 🤔
I ran a script on the existing APIs, and the corresponding types and `variable.kind` are as follows: 

| Type | Kinds |
|--------|--------|
| property | 32 |
| method | 64 |
| type | 256/4194304/128 | 
| event | 32 |

Reply:
It looks like in https://github.com/GoogleChrome/developer.chrome.com/blob/215f8ef57819f451ecf6d5fdd84b9981f60a0a10/external/build/lib/dts-parse.js#L33 we identify events by checking `variable.name` against a list of special types. Would that work?

Reply:
It works! However, I wonder if it's worth implementing the logic again in the script. Will it make understanding or maintenance difficult?

Reply:
I think it's better than looking at the page URLs 😄

Reply:
> I think it's better than looking at the page URLs 😄

Got it, I will try to use `variable.kind` and expect to finish it today.